### PR TITLE
1.0.0-rc.2 release

### PR DESCRIPTION
**Pull request by philvarner**
**Related Issue(s):** 

- n/a

**Proposed Changes:**

1. Release 1.0.0-rc.2 

<https://github.com/radiantearth/stac-api-spec/compare/v1.0.0-rc.1..main>

### Added

- Added optional `numberMatched` and `numberReturned` fields to ItemCollection to align with OGC Commons
  and OAFeat.

### Changed

- The Collections specification has been incorporated into the *Features* specification, but remains as
  a separate conformance class. 
- The Browseable specification has been incorporated into the *Core* specification, but remains as
  a separate conformance class.
- Extensions moved to standalone specification repositories:
    - [Items and Collections API Version](https://github.com/stac-api-extensions/version) 
    - [Fields](https://github.com/stac-api-extensions/fields)
    - [Filter](https://github.com/stac-api-extensions/filter)
    - [Context](https://github.com/stac-api-extensions/context)
    - [Sort](https://github.com/stac-api-extensions/sort)
    - [Transaction](https://github.com/stac-api-extensions/transaction)
    - [Query](https://github.com/stac-api-extensions/query)
    - [Children](https://github.com/stac-api-extensions/children)

### Fixed

- Item Search `limit` parameter semantics now align with OAFeat. The server must not return more Items than the limit and a limit value higher than advertised in the
service description must return a 400 Bad Request status code.

**PR Checklist:**

- [X] This PR has **no** breaking changes.
- [X] This PR does not make any changes to the core spec in the `stac-spec` directory (these are included as a subtree and should be updated directly in [radiantearth/stac-spec](https://github.com/radiantearth/stac-spec))
- [X] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-api-spec/blob/main/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

File: overview.md [hunk 1]
```diff
@@ -123,11 +123,11 @@ have the URIs for conformance to actually resolve to machine-readable informatio
...
-| STAC API - Features    | [Collections and Features](ogcapi-features) | <https://api.stacspec.org/v1.0.0-rc.1/ogcapi-features> | Specifies the use of OGC API - Features to serve STAC Item and Collection objects                               |
-| STAC API - Collections | [Collections and Features](ogcapi-features) | <https://api.stacspec.org/v1.0.0-rc.1/collections>     | Specifies the use of a subset of STAC API - Features to serve Collection objects                                |
+| STAC API - Core        | [Core](core)                                | <https://api.stacspec.org/v1.0.0-rc.2/core>            | Specifies the STAC Landing page `/`, communicating conformance and available endpoints.                         |
+| STAC API - Browseable  | [Core](core)                                | <https://api.stacspec.org/v1.0.0-rc.2/browseable>      | Advertises .                                                                                                    |
```
Comment (cholmes) [inline overview.md:127]: Why does this just say 'Advertises .'?

**Review by cholmes — APPROVED** [review]
Looks good. Left one little comment, that wasn't a change on this PR but was highlighted here.

File: overview.md (hunk 1 above)
Comment (philvarner) [reply overview.md:127]: I apparently stopped typing mid-sentence when I added that. I'll create another PR to fix it before release.

Comment (philvarner) [reply overview.md:127]: Created https://github.com/radiantearth/stac-api-spec/pull/346 to fix that.

Comment (philvarner) [reply overview.md:127]: Resolved.

**Review by matthewhanson — APPROVED** [review]

**Review by jbants — APPROVED** [review]
LGTM

**Review by m-mohr — CHANGES_REQUESTED** [review]
There are still some rc.1 references I think (overview.md)

**Comment by philvarner** [comment]
> There are still some rc.1 references I think (overview.md -> Conformance Class Table)

Odd, not sure why those didn't get replaced. I will fix.

**Comment by philvarner** [comment]
@m-mohr fixed the references in overview.md

**Review by m-mohr — APPROVED** [review]

**Pull request merged**